api/controller.py

```diff
@@ -1406,6 +1406,9 @@ def borrow(self, identifier_type, identifier, mechanism_id=None):
         patron = flask.request.patron
         library = flask.request.library
 
+        header = self.authorization_header()
+        credential = self.manager.auth.get_credential_from_header(header)
+
```

Comment:
This changes behavior slightly in that if you send an invalid request without authentication, you'll now get an authentication failure rather than an "invalid request" error, but you get an error in either case.